testsuite/features/build_validation/init_clients/proxy.feature

```diff
@@ -39,6 +39,10 @@ Feature: Setup containerized proxy
   Scenario: Wait until the proxy host appears
     When I wait until onboarding is completed for "proxy"
 
+  Scenario: Upgrade mgrpxy tool
+    Then I upgrade "proxy" with the last "mgrpxy" version
```

Comment:
Personally, I would simplify this step, by just do an upgrade of all available packages, you can do it via WebUI, or using system.schedulePackageUpdate from the API.

Reply:
Two things here:

Reply:
>  - we should avoid a full upgrade for the potential dependency issues and time consuming action

I don't agree with this, in a Build Validation we must always have the latest released packages, plus, on top, the unreleased packages we are testing.

Otherwise, we will not have the same experience than a customer.


>  - we should use the API for non-testing step to gain in time and stability.
> I would prefer to keep it like that. It also shows what we are doing.

Agree on this, that's why I was proposing the API method here, and in another comment these days.

Reply:
You are right, at first, I though we all agreed on having systems up-to-date when we test on BV, but I was wrong, I clarified it after discussing with more people.
I will add that point as one of our agreements in the confluence page.

About the API, fully agree, and as fallback if we can't do it with API, it's fine doing it with spacecmd, as it is using API behind it.